pkg/controller/labelidentity/label_group_index_test.go

```diff
@@ -0,0 +1,613 @@
+			matchedIDs := i.SetPolicySelectors(tt.selectors, tt.policyKey)
+			assert.ElementsMatch(t, tt.expIDs, matchedIDs)
+			assert.Equalf(t, len(tt.expSelectorItems), len(i.selectorItems.List()),
+				"Unexpected numner of cached selectorItems in step %s", tt.name)
+			for selKey, expSelItem := range tt.expSelectorItems {
+				s, exists, _ := i.selectorItems.GetByKey(selKey)
+				if !exists {
+					t.Errorf("Selector %s is not added in step %s", selKey, tt.name)
+				}
+				sItem := s.(*selectorItem)
+				assert.Truef(t, sItem.policyKeys.Equal(expSelItem.policyKeys),
+					"Unexpected policy keys for selectorItem %s in step %s", selKey, tt.name)
+				assert.Truef(t, sItem.labelIdentityKeys.Equal(expSelItem.labelIdentityKeys),
+					"Unexpected labelIdentity keys for selectorItem %s in step %s", selKey, tt.name)
+			}
+		})
+	}
+}
+
+func TestAddLabelIdentity(t *testing.T) {
+	labelIdentityAOriginalID := uint32(1)
+	tests := []struct {
+		name              string
+		normalizedLabel   string
+		id                uint32
+		originalID        *uint32
+		expPolicyCalled   []string
+		expLabelIdenities map[string]*labelIdentityMatch
+	}{
+		{
+			name:            "Add label identity A",
+			normalizedLabel: labelA,
+			id:              1,
+			expPolicyCalled: []string{"policyA", "policyB"},
+			expLabelIdenities: map[string]*labelIdentityMatch{
+				labelA: {
+					id:               1,
+					selectorItemKeys: sets.NewString(selectorItemA.getKey(), selectorItemB.getKey()),
+				},
+			},
+		},
+		{
+			name:            "Update label identity A",
+			normalizedLabel: labelA,
+			id:              4,
+			originalID:      &labelIdentityAOriginalID,
+			expPolicyCalled: []string{"policyA", "policyB", "policyA", "policyB"},
+			expLabelIdenities: map[string]*labelIdentityMatch{
+				labelA: {
+					id:               4,
+					selectorItemKeys: sets.NewString(selectorItemA.getKey(), selectorItemB.getKey()),
+				},
+			},
+		},
+		{
+			name:            "Add label identity B",
+			normalizedLabel: labelB,
+			id:              2,
+			expPolicyCalled: []string{"policyA", "policyB", "policyD"},
+			expLabelIdenities: map[string]*labelIdentityMatch{
+				labelB: {
+					id:               2,
+					selectorItemKeys: sets.NewString(selectorItemC.getKey(), selectorItemD.getKey()),
+				},
+			},
+		},
+		{
+			name:            "Add label identity C",
+			normalizedLabel: labelC,
+			id:              3,
+			expPolicyCalled: []string{"policyA", "policyB"},
+			expLabelIdenities: map[string]*labelIdentityMatch{
+				labelC: {
+					id:               3,
+					selectorItemKeys: sets.NewString(selectorItemC.getKey()),
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			i := NewLabelIdentityIndex()
+			allSelectorItems := []*selectorItem{
+				selectorItemACopy, selectorItemBCopy, selectorItemCCopy, selectorItemDCopy, selectorItemECopy,
+			}
+			for _, s := range allSelectorItems {
+				i.selectorItems.Update(s)
+			}
```

Comment:
ditto, could it just call `AddSelector` or `SetPolicySelectors` here?

Reply:
Fixed